File: manuallyhandlingruntimechanges/src/main/java/com/codekul/manuallyhandlingruntimechanges/MainActivity.java
package com.codekul.manuallyhandlingruntimechanges;

import android.content.res.Configuration;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        if(newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
            ((ImageView)findViewById(R.id.imgVw)).setImageResource(R.drawable.ic_call_made_black_24dp);
        }
        else if(newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
            ((ImageView)findViewById(R.id.imgVw)).setImageResource(R.drawable.ic_mouse_black_24dp);
        }
        else {
            ((ImageView)findViewById(R.id.imgVw)).setImageResource(R.drawable.ic_launcher_background);
        }
    }
}

File: sharedprefs/src/main/java/com/codekul/sharedprefs/MainActivity.kt
package com.codekul.sharedprefs

import android.content.Context
import android.content.SharedPreferences
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


    }

    fun onRead(view : View) {

        val shPrf = getSharedPreferences("myPrefs", MODE_PRIVATE)
        val nm = shPrf.getString("nm", "none")
        val age = shPrf.getInt("age", -2)
        val isOkay = shPrf.getBoolean("isOkay", false)
        val strength = shPrf.getFloat("strength", -9f)

        Log.i("@codekul", """ Name - $nm Age - $age isOkay - $isOkay Strength - $strength """)
    }

    fun onWrite(view : View) {
        val shPrf = getSharedPreferences("myPrefs", MODE_PRIVATE)
        val edtr = shPrf.edit()
        edtr.putString("nm", "android")
        edtr.putInt("age",15)
        edtr.putBoolean("isOkay", true)
        edtr.putFloat("strength", 45.6f)
        edtr.apply()
    }
}
